lib/train/plugins/base_connection.rb

```diff
@@ -15,18 +15,40 @@ class Train::Plugins::Transport
   class BaseConnection
     include Train::Extras
 
-    # Provide access to the files cache.
-    attr_reader :files
-
     # Create a new Connection instance.
     #
     # @param options [Hash] connection options
     # @yield [self] yields itself for block-style invocation
     def initialize(options = nil)
       @options = options || {}
       @logger = @options.delete(:logger) || Logger.new(STDOUT)
-      @files = {}
       Train::Platforms::Detect::Specifications::OS.load
+
+      # default caching options
+      @cache_enabled = {
+        file: true,
+        command: false,
+      }
+
+      @cache = {}
+      @cache_enabled.each_key do |type|
+        clear_cache(type)
+      end
+    end
+
+    def cache_enabled?(type)
+      @cache_enabled[type.to_sym]
+    end
+
+    # Enable caching types for Train. Currently we support
+    # :file and :command types
+    def enable_cache(type)
+      @cache_enabled[type.to_sym] = true
```

Comment:
Future enhancement: raise an exception if the user tries to enable/disable a cache that we don't support.

Reply:
+1 great catch!